bin/changelogger/WCAdminWriteCommand.php

```diff
@@ -0,0 +1,361 @@
+			return self::FATAL_EXIT;
+		}
+		return self::OK_EXIT;
+	}
+
+	/**
+	 * Delete the change files.
+	 *
+	 * @param InputInterface  $input InputInterface.
+	 * @param OutputInterface $output OutputInterface.
+	 * @param array           $files Files returned from `loadChanges()`.
+	 * @return int
+	 */
+	protected function deleteChanges( InputInterface $input, OutputInterface $output, array $files ) {
+		$dir = Config::changesDir();
+		$ret = self::OK_EXIT;
+		foreach ( $files as $name => $flag ) {
+			if ( $flag >= 2 ) {
+				continue;
+			}
+			Utils::error_clear_last();
+			$ok = quietCall( 'unlink', $dir . DIRECTORY_SEPARATOR . $name );
+			if ( $ok ) {
+				$output->writeln( "Deleted change file $name.", OutputInterface::VERBOSITY_DEBUG );
+			} else {
+				$err = error_get_last();
+				$output->writeln( "<warning>Failed to delete $name: {$err['message']}" );
+				$ret = self::DELETE_FAILED_EXIT;
+			}
+		}
+		return $ret;
+	}
+
+	/**
+	 * Load the changes.
+	 *
+	 * @param InputInterface  $input InputInterface.
+	 * @param OutputInterface $output OutputInterface.
+	 * @return array Array of [ $code, $changes, $files ].
+	 */
+	protected function loadChanges( InputInterface $input, OutputInterface $output ) {
+		$dir = Config::changesDir();
+		if ( ! is_dir( $dir ) ) {
+			$this->askedNoChanges = true;
+			if ( ! $this->askToContinue( $input, $output, 'Changes directory does not exist, so there are no changes to write!' ) ) {
+				return array( self::NO_CHANGE_EXIT, null, null );
+			}
+			return array( self::OK_EXIT, array(), array() );
+		}
+
+		$output->writeln( "Reading changes from $dir...", OutputInterface::VERBOSITY_DEBUG );
+		$files   = null; // Make phpcs happy.
+		$changes = Utils::loadAllChanges( $dir, Config::types(), $this->formatter, $output, $files );
+		$max     = $files ? max( $files ) : 0;
+		if ( $max > 0 && ! $this->askToContinue( $input, $output, ( $max > 1 ? 'Errors' : 'Warnings' ) . ' were encountered while reading changes!' ) ) {
+			return array( self::ASKED_EXIT, null, null );
+		}
+		if ( ! $changes && ! $this->askedNoChanges ) {
+			$this->askedNoChanges = true;
+			if ( ! $this->askToContinue( $input, $output, 'No changes were found!' ) ) {
+				return array( self::NO_CHANGE_EXIT, null, null );
+			}
+		}
+		return array( self::OK_EXIT, $changes, $files );
+	}
+
+	/**
+	 * Deduplicate changes.
+	 *
+	 * @param InputInterface  $input InputInterface.
+	 * @param OutputInterface $output OutputInterface.
+	 * @param Changelog       $changelog Changelog.
+	 * @param ChangeEntry[]   $changes Changes.
+	 * @return int
+	 */
+	protected function deduplicateChanges( InputInterface $input, OutputInterface $output, Changelog $changelog, array &$changes ) {
+		// Deduplicate changes.
+		if ( ! $changes ) {
+			$output->writeln( 'Skipping deduplication, there are no changes.', OutputInterface::VERBOSITY_DEBUG );
+			return self::OK_EXIT;
+		}
+
+		$depth = (int) $input->getOption( 'deduplicate' );
```

Comment:
This is not your code, but I seem to get a `The "deduplicate" option does not exist.` error when I tried to run `npm run changelogger -- write 2.2.1`, it wouldn't work until I added a `return self::OK_EXIT` right above this line.

Did you run into this error as well? am I missing something maybe?

Reply:
This is probably might fault. I've deleted codes from the class and it looks like I removed `deduplicate` option. I will fix it.